projects/wavelet_dataAggregation/checkDataType.py

```diff
@@ -0,0 +1,188 @@
+# Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+from os import listdir
+from os.path import isfile, join, exists
+import csv
+import numpy as np
+from datetime import datetime
+from runDataAggregationExperiment import readCSVfiles
+
+import urllib2
+
+def downloadExampleData():
+  """
+  Download A few example data files from river-view
+  :return: lists of filename, filepath and datatype
+  """
+  exampleDataURL = [
+    'http://data.numenta.org/nypd-motor-vehicle-collisions/nypd-motor-vehicle-collisions/data.csv',
+    'http://data.numenta.org/sfpd-incidents/sfpd-incidents/data.csv',
+    'http://data.numenta.org/usgs-earthquakes/usgs-earthquakes/data.csv',
+    'http://data.numenta.org/portland-911/portland-911/data.csv',
+    'http://data.numenta.org/planetlabs-ephemerides/DOVE%203/data.csv?limit=500',
+    'http://data.numenta.org/nyc-traffic/1/data.csv?limit=500',
+    'http://data.numenta.org/mn-traffic-sensors/10/data.csv?limit=500',
+    'http://data.numenta.org/chicago-beach-water-quality/63rd%20Street%20Beach/data.csv?limit=500'
+  ]
+  dataTypeList = [
+    'transactional',
+    'transactional',
+    'transactional',
+    'transactional',
+    'non-transactional',
+    'non-transactional',
+    'non-transactional',
+    'non-transactional',
+  ]
+  fileNameList = []
+  filePathList = []
+  for url in exampleDataURL:
+    fileName = url.split('/')[3]
+    filePath = 'example_data/' + fileName + '.csv'
+
+    fileNameList.append(fileName)
+    filePathList.append(filePath)
+
+    print "Downloading file " + str(fileName) + " from " + url
+    response = urllib2.urlopen(url)
+
+    data = response.read()
+    rows = data.split('\n')
+    # with open(filePath, 'wb') as f:
+    #     f.write(data)
+    #
+    # fileReader = csv.reader(open(filePath, 'r'))
+    # rows = []
+    # headerRow = fileReader.next() # skip header line
+    # rows.append(headerRow)
+    newRows = []
+    newRows.append(rows[0])
+    for i in range(1, len(rows)-1):
+      row = rows[i].split(',')
+      row[0] = str(datetime.strptime(row[0], '%Y/%m/%d %H:%M:%S'))
+      newRows.append(row)
+
+    fileWriter = csv.writer(open(filePath, 'wb'))
```

Comment:
Use a `with` statement so that the file automatically closes after the nested block of code:
```
    with open(filePath, 'wb') as f:
      fileWriter = csv.writer(f)
      fileWriter.writerows(newRows)

```